lib/pseudodynamic/buffer_raw.cc

```diff
@@ -156,53 +156,26 @@ void pop_buffer_raw::CheckUnPack(int sz) {
  * @return
  */
 bool pop_buffer_raw::Send(pop_combox& s, pop_connection* conn) {
-    // Pack the header (20 bytes)
-    char* dat = packeddata.data();
+    // Pack the header
+    char* dat = get_load();
 
     if (dat == nullptr) {
         LOG_ERROR("fail 1");
         return false;
     }
-
-    int n = packeddata.size();
-    int h[5];
-    memset(h, 0, 5 * sizeof(int));
-
-    int type = header.GetType();
-
-    h[0] = n;
-    h[1] = type;
-
-    switch (type) {
-        case TYPE_REQUEST:
-            h[2] = header.GetClassID();
-            h[3] = header.GetMethodID();
-            h[4] = header.GetSemantics();
-            break;
-        case TYPE_EXCEPTION:
-            h[2] = header.GetExceptionCode();
-            break;
-        case TYPE_RESPONSE:
-            h[2] = header.GetClassID();
-            h[3] = header.GetMethodID();
-            break;
-        default:
-            return false;
-    }
-    memcpy(dat, h, 20);
-
```

Comment:
Check the change just before that.
There is another function, get_load(), which does essentially the same thing as the start of this function did before. In an attempt to reduce code duplication i just call that function.